src/layer/Popup.js

```diff
@@ -420,10 +421,14 @@ Layer.include({
 	// @method openPopup(latlng?: LatLng): this
 	// Opens the bound popup at the specified `latlng` or at the default popup anchor if no `latlng` is passed.
 	openPopup: function (latlng) {
-		if (this._popup && this._popup._prepareOpen(latlng || this._latlng)) {
-
-			// open the popup on the map
-			this._popup.openOn(this._map);
+		if (this._popup) {
+			if (!(this instanceof FeatureGroup)) {
+				this._popup._source = this;
+			}

```

Comment:
I still don't understand it, but I think it has it reason. Can you please add a test for this?

Reply:
I'm not sure what kind of test I could add, but without this if statement there are 4 existing tests that fail. They all seem to relate to feature groups of popups/tooltips with functions as content (in bindTooltip/bindPopup).